readme/Compile.md

```diff
@@ -0,0 +1,47 @@
+# Compilation and Testing
+
```

Comment:
Needs a line or two summarizing what this document is about. Some key points:
* The contents here are mostly for compiling & developing locally.
* For distribution packages, the steps are platform-dependent (which can be found in the `*Build.md` files)